delta/plugins/storage/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/routes/StoragesRoutesSpec.scala

```diff
@@ -0,0 +1,361 @@
+          }
+        }
+      }
+    }
+
+    "fetch a storage original payload" in {
+      val endpoints = List(
+        s"/v1/storages/$uuid/$uuid/remote-disk-storage/source",
+        "/v1/storages/myorg/myproject/remote-disk-storage/source",
+        s"/v1/storages/myorg/myproject/$remoteIdEncoded/source"
+      )
+      forAll(endpoints) { endpoint =>
+        Get(endpoint) ~> routes ~> check {
+          status shouldEqual StatusCodes.OK
+          response.asJson shouldEqual Storage.encryptSource(remoteFieldsJson, config.encryption.crypto).rightValue
+        }
+      }
+    }
+
+    "fetch a storage original payload by rev or tag" in {
+      val endpoints = List(
+        s"/v1/storages/$uuid/$uuid/remote-disk-storage/source",
+        "/v1/storages/myorg/myproject/remote-disk-storage/source",
+        s"/v1/storages/myorg/myproject/$remoteIdEncoded/source"
+      )
+      forAll(endpoints) { endpoint =>
+        forAll(List("rev=1", "tag=mytag")) { param =>
+          Get(s"$endpoint?$param") ~> routes ~> check {
+            status shouldEqual StatusCodes.OK
+            response.asJson shouldEqual Storage.encryptSource(remoteFieldsJson, config.encryption.crypto).rightValue
+          }
+        }
+      }
+    }
+
+    "list storages" in {
+      Get("/v1/storages/myorg/myproject") ~> routes ~> check {
+        status shouldEqual StatusCodes.OK
+        response.asJson shouldEqual jsonContentOf("storage/storages-list.json")
+      }
+    }
+
+    "list not deprecated storages" in {
+      Get("/v1/storages/myorg/myproject?deprecated=false") ~> routes ~> check {
+        status shouldEqual StatusCodes.OK
+        response.asJson shouldEqual jsonContentOf("storage/storages-list-not-deprecated.json")
+      }
+    }
+
+    "fetch the storage tags" in {
+      Get("/v1/storages/myorg/myproject/remote-disk-storage/tags?rev=1") ~> routes ~> check {
+        status shouldEqual StatusCodes.OK
+        response.asJson shouldEqual json"""{"tags": []}""".addContext(contexts.tags)
+      }
+      Get("/v1/storages/myorg/myproject/remote-disk-storage/tags") ~> routes ~> check {
+        status shouldEqual StatusCodes.OK
+        response.asJson shouldEqual json"""{"tags": [{"rev": 1, "tag": "mytag"}]}""".addContext(contexts.tags)
+      }
+    }
+
+    "return not found if tag not found" in {
+      Get("/v1/storages/myorg/myproject/remote-disk-storage?tag=myother") ~> routes ~> check {
+        status shouldEqual StatusCodes.NotFound
+        response.asJson shouldEqual jsonContentOf("/errors/tag-not-found.json", "tag" -> "myother")
+      }
+    }
+
+    "reject if provided rev and tag simultaneously" in {
+      Get("/v1/storages/myorg/myproject/remote-disk-storage?tag=mytag&rev=1") ~> routes ~> check {
+        status shouldEqual StatusCodes.BadRequest
+        response.asJson shouldEqual jsonContentOf("/errors/tag-and-rev-error.json")
+      }
+    }
+
+    "fail to get the events stream without events/read permission" in {
```

Comment:
A passing test for events seems to be missing

Reply:
It is not going to work for now. So far we have to do this on the integration tests, until [this PR is merged on akka-http-testkit](https://github.com/akka/akka-http/pull/3724).

However, I've added tests for the JsonLd encoder part in `StorageEventSpec`.